src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1107,13 +1107,14 @@ private PoolOfThreads.LongRunningWork GetWaitForIndexingThreadToExit(bool disabl
             return null;
         }
 
-        public virtual void Update(IndexDefinitionBase definition, IndexingConfiguration configuration)
+        public virtual void Update(IndexDefinitionBaseServerSide definition, IndexingConfiguration configuration)
         {
             Debug.Assert(Type.IsStatic());
 
             configuration.InitializeAnalyzers(DocumentDatabase.Name);
             InitializeMetrics(configuration);
-            
+            var stateChanged = false;
```

Comment:
Need better name, this isn't just changed, it is changed to `Normal` (but what if this was something else?)

Reply:
If we change to normal we need to start the index. if we change to disable\error we know not to start in StartIndexingThread

Reply:
How do we shut it down, then?

Reply:
we stop the index a few lines before. but I see now I need to add Disable if we change to disable